src/Microsoft.AspNetCore.Razor.Evolution/ITagHelperBoundAttributeDescriptorBuilder.cs

```diff
@@ -165,14 +165,11 @@ private IEnumerable<RazorDiagnostic> Validate()
 
             if (string.IsNullOrWhiteSpace(_name))
             {
-                if (_indexerNamePrefix == null)
```

Comment:
Name can totally be `null` if we have an indexer. I expect this explodes multiple tests.